### PR TITLE
Decide to sign in w/ user identification config (fixes #372)

### LINKED ISSUE #372: Use the payments-config product data to control how the user indentification is handled.
We are currently look at `product.seller.kind === 'donation'` we should use product.user_indentification for that instead.

### PR DESCRIPTION
**Pull request by kumar303**
Fixes https://github.com/mozilla/payments-ui/issues/372

### CODE REVIEW

**Comment by kumar303** [comment]
I can't really write a better test for this until we have https://github.com/mozilla/payments-ui/issues/353 (the sign-in logic is currently tested though)

File: public/js/views/transaction/product-pay.jsx [hunk 1]
```diff
@@ -11,7 +11,6 @@ import { gettext } from 'utils';
 export default class ProductPay extends Component {
 
   static propTypes = {
-    // Amount to pay, which only applies to things like donations.
```
Comment (muffinresearch) [inline public/js/views/transaction/product-pay.jsx]: Thanks for cleaning this up.

**Comment by muffinresearch** [comment]
r+wc

**Comment by kumar303** [comment]
> Was there other places the donations condition was checked. 

There are but I think they make sense as is because they are actually about donations and not about user identification:

https://github.com/mozilla/payments-ui/blob/master/public/js/views/transaction/complete-payment.jsx#L70
https://github.com/mozilla/payments-ui/blob/master/public/js/views/transaction/product-pay.jsx#L45
https://github.com/mozilla/payments-ui/blob/master/public/js/views/transaction/product-pay-chooser.jsx#L35

**Pull request merged**